#!/bin/sh -e
## 18_donotopen_hda.dpatch.dpatch by Andreas Metzler <ametzler@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: dev=ATA:1,0,0 uselessly opens /dev/hda, breaking non-root access.
## DP: See #228215

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /tmp/cdrtools-2.01/libscg/scsi-linux-sg.c cdrtools-2.01/libscg/scsi-linux-sg.c
--- /tmp/cdrtools-2.01/libscg/scsi-linux-sg.c	2004-05-20 15:42:12.000000000 +0200
+++ cdrtools-2.01/libscg/scsi-linux-sg.c	2004-06-12 22:08:56.000000000 +0200
@@ -391,7 +391,10 @@
 	 * look silly but there may be users that did boot from a SCSI hdd
 	 * and connected 4 CD/DVD writers to both IDE cables in the PC.
 	 */
-	if (use_ata) for (i = 0; i <= 25; i++) {
+/*	if (use_ata) for (i = 0; i <= 25; i++) { */
+/* If a device was specified with ATA:x,y,z try to open this device instead of
+ * uselessly opening all of them until we reach the specified one */
+	if (use_ata) for (i=2*busno+tgt >= 0 ? 2*busno+tgt:0; i <= 25; i++) {
 		js_snprintf(devname, sizeof (devname), "/dev/hd%c", i+'a');
 					/* O_NONBLOCK is dangerous */
 		f = open(devname, O_RDWR | O_NONBLOCK);
